test/functional/test_framework/test_node.py

```diff
@@ -198,6 +204,8 @@ def start(self, extra_args=None, *, cwd=None, stdout=None, stderr=None, env=None
         if extra_args is None:
             extra_args = self.extra_args
 
+        self.use_v2transport = "-v2transport=1" in extra_args or (self.default_to_v2 and "-v2transport=0" not in extra_args)
```

Comment:
68a9001: there's a behaviour difference when test is run with `--v2transport` option and `TestNode`(initially `"v2transport=0"`) is restarted with:
- `extra_args` supplied => `TestNode` restarts as v2 node
- `extra_args` not supplied => `TestNode` restarts as v1 node

maybe when `extra_args` is `None`, we could remove `"v2transport"` stuff it gets from the old `self.extra_args` using something like `extra_args = list(filter(lambda s: not s.startswith("-v2transport="), extra_args))`?

Reply:
One small drawback of the current way of inspecting `extra_args` for the v2transport option is that it doesn't work if the argument is specified in a slightly different format with same meaning. For example, `-v2transport=0`/`-v2transport=1` could also be passed as `-nov2transport`/`-v2transport`, `-nov2transport=1`/`-nov2transport=0` (admittedly, no sane person would use the last one though). It might be worth it to consider that in a follow-up, e.g. by enforcing with an assertion that arguments that contain the string "v2transport" must be prefixed by "-" and suffixed by either "=0" or "=1". As far as I can see, that's the first time that the test framework needs to inspect bitcoind arguments, so that's a new problem. 

@stratospher: I agree that this is a bit counter-intuitive, but it's in line with the current behavior of `restart_node` (that apparently many tests rely on), i.e. "if no extra_args are passed, restart with the ones that were initially specified (`self.extra_args`)", in your example `-v2transport=0`.

Reply:
> "if no extra_args are passed, restart with the ones that were initially specified (self.extra_args)"

oh interesting! so https://github.com/bitcoin/bitcoin/pull/28805#discussion_r1388909333 is out of scope for this PR.